### PR TITLE
LLaVaNeXT: pad on right if training

### LINKED ISSUE #32112: LlavaNextVideo always assumes left padding when batch size is 1
### System Info

Unrelated to this issue

### Who can help?

@zucchini-nlp @Narsil 

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

At the moment I don't have a concise script for reproducing, but I think the problem should be clear given the below description. 

Basically I'm trying to finetune the new LlavaNextVideo model. The current implementation (see code below) will always assume `left padding` when the batch size is 1. This will cause a "index out of bound" error when training with batch size being 1, due to the unexpected left padding (instead of right padding) for training.

https://github.com/huggingface/transformers/blob/0fdea8607d7e01eb0e38a1ebeb7feee30a22f0cf/src/transformers/models/llava_next_video/modeling_llava_next_video.py#L565-L576

I can try to put up an example with concrete shape/dims to better show the behavior, but just on a high level I believe the desired behavior is that as long as we are in the training mode (determined by the `_left_padding` and `_right_padding`), we should set `left_padding` to False, regardless of whether batch size is 1.

### Expected behavior

https://github.com/huggingface/transformers/blob/0fdea8607d7e01eb0e38a1ebeb7feee30a22f0cf/src/transformers/models/llava_next_video/modeling_llava_next_video.py#L565-L576

As long as we are in the training mode (determined by the `_left_padding` and `_right_padding`), we should set `left_padding` to False, regardless of whether batch size is 1.

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes #32112. The issue didn't mention any bug, as it's using batch-size=1 for training but it led me to this PR. This way we are more confident on which side to pad the inputs. I verified the padding is "right" when training with Trainer and in inference it runs to the block where we infer padding from attn mask

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for digging into this and fixing! 

Could you add a quick test which checks the padding is set as expected for the different modes?

**Comment by zucchini-nlp** [comment]
Added a test @amyeroberts

File: docs/source/en/model_doc/llava-next-video.md [hunk 1]
```diff
@@ -43,6 +43,13 @@ The original code can be found [here](https://github.com/LLaVA-VL/LLaVA-NeXT/tre
 
 - We advise users to use `padding_side="left"` when computing batched generation as it leads to more accurate results. Simply make sure to call `processor.tokenizer.padding_side = "left"` before generating.
 
+<Tip warning={true}>
+
+- Llava-Next uses different number of patches for images and thus has to pad the inputs inside modeling code, aside from the padding done when processing the inputs. The default setting is "left-padding" if model is in `eval()` mode, otherwise "right-padding".
```
Comment (amyeroberts) [inline docs/source/en/model_doc/llava-next-video.md:48]: Nice!

**Review by amyeroberts — COMMENTED** [review]
Looks great - thanks for handling and adding the test!

**Review by amyeroberts — APPROVED** [review]
